### PR TITLE
Add a feature for the presence of the gfx940 MFMAs

### PR DESCRIPTION
**Pull request by krzysz00**
This feature currently isn't hooked up to anything, but at least it's there so we can use it in the future.

### CODE REVIEW

**Comment by manupak** [comment]
Overall changes are fine!.

Just naming question : why not use mfma_i8 and mfma_xf32 ? 

if not, Is it more of a random assortment of intrinsics?

**Comment by krzysz00** [comment]
It's very much a random assortment. We have FP8, changes to I8, XF32, a bunch of old BF16's being deprecated ...

File: mlir/lib/Dialect/Rock/Generator/AmdArchDb.cpp [hunk 1]
```diff
@@ -39,7 +41,8 @@ AmdArchInfo mlir::rock::lookupArchInfo(StringRef arch) {
   StringRef major = chip.slice(0, chip.size() - 2);
   if (major == "gfx9") {
     return llvm::StringSwitch<AmdArchInfo>(minor)
-        .Cases("08", "0a", "40", cdnaInfo)
+        .Cases("08", "0a", cdnaInfo)
+        .Case("40", cdna3Info)
```
Comment (manupak) [inline mlir/lib/Dialect/Rock/Generator/AmdArchDb.cpp:45]: Please mirror this change in mlir/test/lit.site.cfg.py.in:set_arch_features()

**Review by manupak — COMMENTED** [review]
Please adjust lit cfg accordingly. 

Re: naming,
oh dear -- I dont have a better solution either given the bit enum format of features we use.
(An alternative might've been to use a hierarchical target structure -- but thats a larger change)

File: mlir/lib/Dialect/Rock/Generator/AmdArchDb.cpp (hunk 1 above)
Comment (krzysz00) [reply mlir/lib/Dialect/Rock/Generator/AmdArchDb.cpp:45]: Right, good point.

Part of me is thinking we want to bring the arch string back ...

Ok, yeah, actually, given the _rest_ of the MFMA changes, I'm bringing the arch string back.

**Comment by jerryyin** [comment]
I'm confused about this PR. It sounds like in #1013 you have already decided to add the architecture field in the mfma selection. So why not go one step further and encode it in the mfma map itself? Then at selection time it will be straightforward to check if the mfma is supported by the architecture encoded in the map entry.

The approach you are taking in this PR seems to suggest we'll represent all those in a separate way using the high level gemm features. In my opinion, it goes in conflict with #1013.

**Comment by krzysz00** [comment]
Yeah, this is supersceded by #1013 so closing

**Pull request closed without merging**